atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -206,7 +206,7 @@ protected SnapshotTransaction createTransaction(long immutableLockTs,
                 cleaner.getTransactionReadTimeoutMillis(),
                 TransactionReadSentinelBehavior.THROW_EXCEPTION,
                 allowHiddenTableAccess);
-        return runTaskThrowOnConflict(task, new OnlyWriteTempTablesTransaction(transaction, sweepStrategyManager));
+        return runTaskThrowOnConflict(task, transaction);
```

Comment:
I think this is the line causing your test failures, I suspect `transaction` here needs to be wrapped to make this a read only transaction